File: Backend/src/main/java/com/example/Inventra/service/ProductService.java
package com.example.Inventra.service;

import com.example.Inventra.dto.ProductCreateRequest;
import com.example.Inventra.dto.ProductUpdateRequest;
import com.example.Inventra.dto.ProductResponseDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.List;

public interface ProductService {
    ProductResponseDTO createProduct(ProductCreateRequest request);
    ProductResponseDTO updateProduct(Long id, ProductUpdateRequest request);
    ProductResponseDTO getProductById(Long id);
    ProductResponseDTO getProductBySku(String sku);
    Page<ProductResponseDTO> getAllProducts(Pageable pageable);
    List<ProductResponseDTO> getLowStockProducts();
    void deactivateProduct(Long id);
}

File: frontend/inventra-frontend/src/services/productService.js
import api from './api';

export const getProducts = (page = 0, size = 10) =>
  api.get(`/products/get-all-products?page=${page}&size=${size}`);

export const getProductById = (id) => api.get(`/products/id/${id}`);

export const getProductBySku = (sku) => api.get(`/products/sku/${sku}`);

export const createProduct = (data) => api.post('/products/create', data);

export const updateProduct = (id, data) => api.put(`/products/id/${id}`, data);

export const deleteProduct = (id) => api.delete(`/products/id/${id}`);